Parse/ParseClientConfiguration.h

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Parse/PFConstants.h>
+
+NS_ASSUME_NONNULL_BEGIN
+
+@protocol ParseMutableClientConfiguration<NSObject>
+
+@property (nonatomic, nullable, copy) NSString *applicationId;
+@property (nonatomic, nullable, copy) NSString *clientKey;
+
+@property (nonatomic, assign, getter=isLocalDatastoreEnabled) BOOL localDatastoreEnabled;
+
+@property (nonatomic, nullable, copy) NSString *applicationGroupIdentifier;
+@property (nonatomic, nullable, copy) NSString *containingApplicationBundleIdentifier;
+
+@end
+
+/*!
+ This represents the entirety of a parse configuration
+ */
+@interface ParseClientConfiguration : NSObject<NSCopying>
+
++ (instancetype)new NS_UNAVAILABLE;
++ (instancetype)alloc NS_UNAVAILABLE;
```

Comment:
I don't think we need these, simply due to the fact that you can't `init` it anyway after allocating... 2 less methods polluting the name space?